File: reverse-integer/reverse_integer_test.go
package reverse

import "testing"

func TestReverse(t *testing.T) {
	testCases := []struct {
		x    int32
		want int32
	}{
		{x: 123, want: 321},
		{x: -123, want: -321},
		{x: 0, want: 0},
		{x: 123000, want: 321},
		{x: 2147483647, want: 0},
	}

	for idx, testCase := range testCases {
		if reverseInteger(testCase.x) != testCase.want {
			t.Errorf("%d is failed.", idx)
		}
	}
}

File: reverse-integer/reverse_integer_test.py
import unittest

from reverse_integer import Solution

class TestReverse(unittest.TestCase):
  def setUp(self):
    self.testCases =  [
      {
        'x': 123,
        'want': 321,
      },
      {
        'x': -123,
        'want': -321,
      },
      {
        'x': 120,
        'want': 21,
      },
      {
        'x': 0,
        'want': 0,
      },
      {
        'x': -1234000100,
        'want': -10004321,
      },
      {
        'x': 99999999999999999,
        'want': 0,
      },
    ]

  def testRev(self):
    sol = Solution()
    for testCase in self.testCases:
      with self.subTest(testCase=testCase):
        res = sol.reverse(testCase['x'])
        self.assertEqual(res, testCase['want'])

if __name__ == '__main__':
  unittest.main()